recursion/gnark-ffi/src/groth16.rs

```diff
@@ -26,9 +33,46 @@ pub struct Groth16Proof {
 }
 
 impl Groth16Prover {
-    /// Creates a nejw verifier.
+    /// Starts up the Gnark server using Groth16 on the given port.
     pub fn new() -> Self {
-        Groth16Prover
+        let manifest_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+        let gnark_dir = manifest_dir.join("../gnark");
+        let port = env::var("HOST_PORT").unwrap_or_else(|_| "8080".to_string());
+        let port_clone = port.clone();
+
+        // Start the Gnark server in a separate thread.
+        let rt = Runtime::new().unwrap();
+        rt.block_on(async {
+            tokio::spawn(async move {
+                let mut child = Command::new("go")
+                    .args([
+                        "run",
+                        "main.go",
+                        "serve",
+                        "--bucket",
```

Comment:
I don't think the AWS bucket stuff should be handled by the Go script.

Reply:
you mean exposong it as a flag?

Reply:
removed the flag